app/assets/javascripts/student_profile_v2/chart.js

```diff
@@ -0,0 +1,77 @@
+(function() {
+  window.shared || (window.shared = {});
+  var dom = window.shared.ReactHelpers.dom;
+  var createEl = window.shared.ReactHelpers.createEl;
+  var merge = window.shared.ReactHelpers.merge;
+
+  var ProfileChartSettings = window.ProfileChartSettings;
+  var HighchartsWrapper = window.shared.HighchartsWrapper;
+
+  // Component for all charts in the profile page.
+  window.shared.Chart = React.createClass({
+    displayName: 'Chart',
+
+    propTypes: {
+      series: React.PropTypes.arrayOf( // you can plot multiple series on the same graph
+        React.PropTypes.shape({
+          name: React.PropTypes.string.isRequired, // e.g. 'Scaled score'
+          data: React.PropTypes.array.isRequired // [year, month, date, value] quads
+        })
+      ),
+      title: React.PropTypes.string.isRequired, // e.g. 'MCAS scores, last 4 years'
+      yAxis: React.PropTypes.object.isRequired // options for rendering the y-axis
+    },
+
+    quadsToPairs: function(quads) {
+      return quads.map(function(quad) {
+        var year = quad[0],
+            month = quad[1],
+            day = quad[2],
+            value = quad[3];
+        return [Date.UTC(year, month - 1, day), value]; // one-index --> zero-index on month
+      });
+    },
+
+    baseOptions: function() {
+      return merge(ProfileChartSettings.base_options, {
+        xAxis: merge(ProfileChartSettings.x_axis_datetime, {
+          plotLines: this.x_axis_bands,
+          min: this.props.timestampRange.min,
+          max: this.props.timestampRange.max
+        })
+      });
+    },
+
+    getDefaultProps: function() {
+      var now = moment.utc().toDate();
+      // TODO(kr) align to school year?
+      // The intent of fixing this date range is that when staff are looking at profile of different students,
+      // the scales are consistent (and not changing between 3 mos and 6 years depending on the student's record,
+      // since that's easy to miss and misinterpret.
+      var intervalBack = [4, 'years'];
+
+      return {
+        now: now,
+        intervalBack: intervalBack,
+        timestampRange: {
+          min: moment(now).subtract(4, 'years').toDate().getTime(),
+          max: now.getTime(),
+        }
+      };
+    },
+
+    render: function() {
+      var self = this;
```

Comment:
Alternately you can pass the `this` value to `.map` like so: `this.props.series.map(function(obj) { ... }, this)`.  Cool either way.